### PR TITLE
Internal refactoring for the add_rule rework

### PR DESCRIPTION
**Pull request by rusty-snake**
- simplify add_rule
- Make ScmpArgCompare be just a newtype wrapper around scmp_arg_cmp with
  the same ABI.
- Add unit test for ScmpArgCompare

### CODE REVIEW

**Comment by ManaSugi** [comment]
LGTM! Could you resolve a conflict?

**Comment by rusty-snake** [comment]
Depends on #36.

**Comment by ManaSugi** [comment]
Could you create unit tests for this to avoid decreasing test code coverage?

File: libseccomp/src/lib.rs [hunk 1]
```diff
@@ -1086,6 +1059,28 @@ mod tests {
         assert_eq!(expected_api, api);
     }
 
+    #[test]
+    fn test_scmpargcompare() {
```
Comment (ManaSugi) [inline libseccomp/src/lib.rs:1063]: Could you add or replace tests for `From` trait as follows? ( currently, the trait is not covered by tests)

```rust
 assert_eq!(                                                                 
     scmp_arg_cmp::from(ScmpArgCompare::new(0, ScmpCompareOp::NotEqual, 8)), 
     scmp_arg_cmp {                                                          
         arg: 0,                                                             
         op: scmp_compare::SCMP_CMP_NE,                                      
         datum_a: 8,                                                         
         datum_b: 0,                                                         
     }                                                                       
 );                                                                          
 assert_eq!(                                                                 
     scmp_arg_cmp::from(&ScmpArgCompare::new(0, ScmpCompareOp::NotEqual, 8)),
     scmp_arg_cmp {                                                          
         arg: 0,                                                             
         op: scmp_compare::SCMP_CMP_NE,                                      
         datum_a: 8,                                                         
         datum_b: 0                                                          
     }                                                                       
 );
```

**Comment by ManaSugi** [comment]
Thank you! please squash the commits.

**Comment by ManaSugi** [comment]
LGTM. Thanks!

**Review by ManaSugi — APPROVED** [review]

**Pull request merged**